### PR TITLE
Add Spring Boot support

### PR DESCRIPTION
**Pull request by philwebb**
Provide support classes and auto-configuration to allow Atomikos to be
used with Spring Boot. This commit introduces two user-facing modules
along with an integration test module that doesn't need to be published.

The `transactions-spring-boot` class provides support classes along
with auto-configuration. Most of the code was originally developed in
Spring Boot, however, going forward it will be maintained directly here.

The `transactions-spring-boot-starter` provides a starter POM equivalent
to the `spring-boot-starter-jta-atomikos` POM that ships with Spring
Boot 2.3.

_Write your description here. BTW we're really honoured to receive your help, but please understand that some pull requests may be rejected for several reasons - in particular because it might conflict with the commercial offerings of Atomikos. Without making money on something, we would not have been able to give TransactionsEssentials for free in the first place! Thanks for understanding..._

### CODE REVIEW

**Review by GuyPardon — COMMENTED** [review]
Thanks Phil!

Concerning the properties: do you think it is technically feasible to support the existing property prefix for easy migration purposes? Just to avoid we have to answer thousands of times the same upgrade questions :-) 

For instance: we could log a warning if we find prefixes like "spring.jta" but for now accept these and use them if found in the configuration?

**Comment by philwebb** [comment]
@GuyPardon I suspect it might be a little tricky since the `spring.jta` properties will still be bound to Spring Boot classes in 2.4. I'll need to investigate.

**Comment by GuyPardon** [comment]
OK thanks!

**Comment by GuyPardon** [comment]
@philwebb Just curious: did you get any input on the properties?

**Comment by philwebb** [comment]
Sorry, not yet. I had Thanksgiving off and now have quite a backlog.

File: public/pom.xml [hunk 1]
```diff
@@ -29,6 +29,9 @@
 		<module>transactions-jndi-provider</module>
 		<module>atomikos-transactions-essentials</module>
 		<module>transactions-remoting</module>
+		<module>transactions-spring-boot</module>
+		<module>transactions-spring-boot-starter</module>
```
Comment (linghengqian) [inline public/pom.xml:33]: So, what is the relationship between this PR and https://www.atomikos.com/Blog/TransactionsEssentials5dot0dot9#Feature188756_Add_spring_boot_starter ?

**Pull request closed without merging**